src/main/java/ch/wisv/areafiftylan/seats/service/SeatServiceImpl.java

```diff
@@ -52,80 +52,73 @@ public SeatServiceImpl(SeatRepository seatRepository, TeamService teamService, T
 
     }
 
     @Override
-    public SeatmapResponse getAllSeats() {
-        List<Seat> all = seatRepository.findAll();
-
-        Map<String, List<Seat>> seatMapGroups = all.stream().collect(Collectors.groupingBy(Seat::getSeatGroup));
+    public List<Seat> getSeatsByTeamName(String teamName) {
+        Team team = teamService.getTeamByTeamname(teamName);
 
-        return new SeatmapResponse(seatMapGroups);
+        return team.getMembers().stream().
+                map(User::getEmail).
+                map(seatRepository::findByTicketOwnerEmailIgnoreCase).
+                flatMap(Collection::stream).
+                collect(Collectors.toList());
     }
 
     @Override
-    public SeatmapResponse getSeatGroupByName(String groupname) {
-        List<Seat> seatGroup = seatRepository.findBySeatGroup(groupname);
+    public SeatmapResponse getSeatGroupByName(String groupName) {
+        List<Seat> seatGroup = seatRepository.findBySeatGroup(groupName);
 
         Map<String, List<Seat>> seatMapResponse = new HashMap<>();
-        seatMapResponse.put(groupname, seatGroup);
+        seatMapResponse.put(groupName, seatGroup);
 
         return new SeatmapResponse(seatMapResponse);
     }
 
     @Override
-    public boolean reserveSeatForTicket(String seatGroup, int seatNumber, Long ticketId) {
-        Seat seat = getSeatBySeatGroupAndSeatNumber(seatGroup, seatNumber);
-        if (seat.isTaken()) {
-            return false;
-        } else {
-            reserveSeat(seat, ticketId);
-            return true;
-        }
+    public Seat getSeatBySeatGroupAndSeatNumber(String groupName, int seatNumber) {
+        return seatRepository.findBySeatGroupAndSeatNumber(groupName, seatNumber)
+                .orElseThrow(SeatNotFoundException::new);
     }
 
     @Override
-    public void reserveSeatForAdmin(String seatGroup, int seatNumber, Long ticketId) {
-        Seat seat = getSeatBySeatGroupAndSeatNumber(seatGroup, seatNumber);
-        reserveSeat(seat, ticketId);
-    }
-
-    /**
-     * Reserves a Seat, and sets the Ticket to ticketId's Ticket.
-     * Also frees any previous Seat belonging to the Ticket.
-     * This method ignores any current user assigned to the seat, that logic must be done before calling this class.
-     * If any user was assigned this seat before, send an email to let them know their seat was overridden.
-     * This method is synchronized to prevent the extremely unlikely event of simultaneous requests.
-     *
-     * @param seat The Seat to reserve.
-     * @param ticketId The Id of the Ticket to be set to the Seat.
-     */
     @Synchronized
-    private void reserveSeat(Seat seat, Long ticketId) {
+    public boolean reserveSeat(String groupName, int seatNumber, Long ticketId, boolean allowSeatOverride) {
+        Seat seat = getSeatBySeatGroupAndSeatNumber(groupName, seatNumber);
+        Ticket ticket = null;
+
+        if (!allowSeatOverride && (seat.isTaken() || seat.isLocked() || ticketId == null)) {
```

Comment:
Shouldn't it be `ticketId != null` since we are checking for not allowed seat modifications?